### DISCUSSION TITLE
Projenin Mobil Uygulamaya Dönüştürülmesi

### DISCUSSION THREAD

**Discussion by cagrimertbakirci** [evrimagaci/periodum#42] [Ideas]
Şu anda bunun için epey erken olabilir; ama fikirleri toplamak ve olasılıkları tartışmak açısından açmakta fayda görüyorum: Her ne kadar çok örneği olsa da, bu kadar detaylı bir veritabanına yaslanan Türkçe ve interaktif bir periyodik tablo olmamasından ötürü, acaba ayrı bir Android/iOS uygulama da yapsak mı

**Artıları:**
- Çok daha geniş kitleye ulaşabilmesi.
- Mobil siteden erişimi direkt olarak app'e yönlendirebiliriz.
- Veri tüketimini azaltabiliriz.

**Eksileri:**
- Çok daha fazla yük getirecek olması (hem mobil hem web tasarlanacak; yeterli komünite desteği gelirse sorun olmaz)

Zamanla bu listeyi genişleterek nihai kararı verebiliriz. Şimdilik böylece dursun.

**Comment by fresholia** [comment]
Bu noktada ilk belirleyici nokta kullanılacak teknoloji olacak. Uygulama için bana göre iki ana alternatifimiz var, bunlar ise React Native ve Flutter. Topluluğumuzun bilgi düzeyine göre ikisinden birini seçebiliriz, eksilerini ve artılarını tartışabiliriz. Bunun için GitHub'un Discussions bölümünü kullanabiliriz.

**Comment by fzerman** [comment]
Burada gerek ana proje dilinin JavaScript olması gerek bilen kişi sayısının fazla olması nedeni ile react native seçilebilir. Ayriyeten, eğer web uygulaması baştan sona responsive tasarlanırsa direk progressive web app'de seçilebilir. Böylece, yazılımcıları mobil ve web olarak ayırmaktansa tamamını webe kanalize edebiliriz.

**Comment by dyatlov86** [comment]
flutterda yazarsak ios/android/web olarak 1 taşla 3 kuş vurabiliriz.

https://docs.flutter.dev/development/tools/sdk/release-notes/supported-platforms

**Comment by akifbora** [comment]
pwa desteği eklenebilir. böylece sadece mobile değil masaüstüne de indirilebilir.

**Comment by birkanozer** [comment]
Projeyi komple React a geçirebilirsek mobile entegre etmesi ve takibi çok daha kolay olur. Ortak komponentleri (element kutuları, bilgi kutuları, sıcaklık çubuğu) ortak repoda toplarız. Hatta bu işe spesifik yeni bir UI repo da yazabiliriz. Komponentleri mobile first mantığında React a aktarsak mobil dönüşümün büyük kısmı hallolmuş olur.

**Comment by nesimtunc** [comment]
Bununla ilgili calisma baslatildi. https://github.com/evrimagaci/periodum-apple

**Comment by firatgursoy** [comment]
Native yerine basit bir webview bu işi görebilir. Performans olarak ufak bir fark olsa da; yeni geliştirmelerin diğer projelere aktarılması gereksiz efora ve takip edilmezse uygulamaların birbirinden farklı olmasına sebep olur.

**Reply by fresholia** [reply]
Apple Webview uygulamalarını genellikle kabul etmiyor
